tools/rpkg/tests/testthat/test_struct.R

```diff
@@ -0,0 +1,110 @@
+test_that("structs can be read", {
+  skip_if_not_installed("vctrs")
+
+  con <- dbConnect(duckdb::duckdb())
+  on.exit(dbDisconnect(con, shutdown = TRUE))
+
+  res <- dbGetQuery(con, "SELECT {'x': 100, 'y': 'hello', 'z': 3.14} AS s")
+  expect_equal(res, vctrs::data_frame(
```

Comment:
Here and below, the tests use vctrs and tibble, are both required or could one of them (or base R even) suffice?

Reply:
Constructing nested and packed data frames is ugly in base R, it requires the use of `I()` which will break the subsequent comparison and also creates non-integer row names in one case:

``` r
dput(data.frame(a = I(data.frame(b = 1, c = 2))))
#> structure(list(a = structure(list(b = 1, c = 2), class = c("AsIs", 
#> "data.frame"), row.names = c(NA, -1L))), class = "data.frame", row.names = "1")
dput(data.frame(a = I(list(data.frame(b = 1:2, c = 2:3)))))
#> structure(list(a = structure(list(structure(list(b = 1:2, c = 2:3), class = "data.frame", row.names = c(NA, 
#> -2L))), class = "AsIs")), class = "data.frame", row.names = c(NA, 
#> -1L))
```

<sup>Created on 2022-06-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

If we don't want to bring in vctrs, we can hand-roll a constructor that doesn't have this limitation, but read on.

The arrow package returns tibbles and "list_of" vectors. Implementing the comparison without the tibble package (which, like arrow, imports vctrs), will require substantial workarounds.

Reply:
Ok thanks for the explanation